src/block/block_read.c

```diff
@@ -221,7 +221,21 @@ __wt_block_read_off(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_ITEM *buf, uin
     check_size = F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP;
     if (swap.checksum == checksum) {
         blk->checksum = 0;
-        if (__wt_checksum_match(buf->mem, check_size, checksum)) {
+
+        /*
+         * If chunk cache is configured we want to account for the race condition where the chunk
+         * cache could have stale content, and therefore a mismatched checksum. We do not want to
+         * fail here, free the stale content and read once so we can retry.
+         */
+        checksum_match = __wt_checksum_match(buf->mem, check_size, checksum);
+        if (!checksum_match) {
+            if (F_ISSET(&S2C(session)->chunkcache, WT_CHUNKCACHE_CONFIGURED)) {
+                WT_RET(__wt_chunkcache_free_external(session, block, objectid, offset, size));
```

Comment:
I think we should have a way to know when this happens (i.e., when we have to reload data because the chunk cache content is incorrect). For example, suppose a customer has something wrong with the storage they use for the chunk cache. If we just silently reload data over-and-over, they complain about performance and we don't have an easy way to see what's going on.

Not sure if the best option is a stat, a verbose message, or a `wt_errx()` message or a combination. What do you all think, @ruby-oujo & @korteland?

Reply:
The stats are in a separate ticket (WT-11726). I've added this feedback on that ticket, but I agree we should add some logging here.

Reply:
I opted for a verbose message - deferring stats to the other ticket. [d39d104](https://github.com/wiredtiger/wiredtiger/pull/9838/commits/d39d104c6dd7aa195dd49d17d126e14adffa9a52)

Reply:
That works for me. Thanks!